src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -165,12 +166,15 @@ internal Authority GetAuthority(IAccount account)
                 customAuthority,
                 cache,
                 commonParameters,
-                CreateRequestContext());
+                requestContext);
         }
 
-        private RequestContext CreateRequestContext()
+        // This implementation should ONLY be called for cases where we aren't participating in
+        // MATS telemetry but still need a requestcontext/logger, such as "GetAccounts()".
+        // For service calls, the request context should be created in the **Executor classes as part of request execution.
+        private RequestContext CreateRequestContext(Guid telemetryCorrelationId)
```

Comment:
>Guid telemetryCorrelationId [](start = 52, length = 27)

is it required to pass this in? can't you create the new guid in this class?

Reply:
no, it's required to pass this in since as noted above the id to correlate an acquiretoken loop is created very early so that all logging/telemetry can be associated with it.  so we want to use the same value throughout the stack.

---
In reply to: [264540440](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/955#discussion_r264540440) [](ancestors = 264540440)